core/src/main/scala/kafka/log/Log.scala

```diff
@@ -582,6 +582,64 @@ class Log(val dir: File,
     }
   }
 
+  private class SizeBasedRetentionCheck(var diff: Long) {
```

Comment:
Got it. The alternative is to create an anonymous function, but probably less readable. Maybe worth making `diff` private.